evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -2794,3 +2794,141 @@ def test_partial_dependence_fast_mode_errors_if_train(
+    pipeline = BinaryClassificationPipeline(
+        [dfs_transformer, "Standard Scaler", "Random Forest Classifier"],
+    )
+
+    # Engineered features have the their origins specified as either "base" or "engineered"
+    # it has to remain set for partial dependence to be able to predict on the updated data
+    engineered_feature = "ABSOLUTE(1)"
+    assert X_fm.ww.columns[engineered_feature].origin == "engineered"
+
+    pipeline.fit(X_fm, y)
+    part_dep = partial_dependence(
+        pipeline,
+        X_fm,
+        features=engineered_feature,
+        grid_resolution=2,
+        fast_mode=fast_mode,
+        X_train=X_fm,
+        y_train=y,
+    )
+
+    assert not part_dep.feature_values.isna().any()
+    assert not part_dep.partial_dependence.isna().any()
+
+
+@pytest.mark.parametrize("fast_mode", [True, False])
+def test_partial_dependence_dfs_transformer_handling_with_multi_output_primitive(
+    fast_mode,
+    df_with_url_and_email,
+):
+    X = df_with_url_and_email
+    y = pd.Series(range(len(X)))
+    X.ww.name = "X"
+    X.ww.set_index("numeric")
+    X.ww.set_types(logical_types={"categorical": "NaturalLanguage"})
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="X",
+        dataframe=X,
+        index="index",
+        make_index=True,
+    )
+    X_fm, features = ft.dfs(
+        entityset=es,
+        target_dataframe_name="X",
+        trans_primitives=["LSA"],
+    )
+
+    dfs_transformer = DFSTransformer(features=features)
+    pipeline = RegressionPipeline(
+        [dfs_transformer, "Standard Scaler", "Random Forest Regressor"],
+    )
+    # Confirm that the LSA primitive was actually used
+    assert any(len(f.get_feature_names()) > 1 for f in features)
+
+    pipeline.fit(X_fm, y)
+    part_dep = partial_dependence(pipeline, X_fm, features=1, grid_resolution=5)
+    fast_part_dep = partial_dependence(
+        pipeline,
+        X_fm,
+        features=1,
+        grid_resolution=5,
+        fast_mode=fast_mode,
+        X_train=X_fm,
+        y_train=y,
+    )
+    pd.testing.assert_frame_equal(part_dep, fast_part_dep)
+
+
+@pytest.mark.parametrize("fast_mode", [True, False])
+def test_partial_dependence_dfs_transformer_target_in_features(fast_mode, X_y_binary):
+    X, y = X_y_binary
+    X = pd.DataFrame(X)
+    X.columns = X.columns.astype(str)
+
+    # Insert y into X so that it's part of the EntitySet
+    # and then ignore in DFS later on so it's not in X_fm
+    X["target"] = y
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="X",
+        dataframe=X,
+        index="index",
+        make_index=True,
+    )
+    seed_features = [ft.Feature(es["X"].ww["target"])]
+    X_fm, features = ft.dfs(
+        entityset=es,
+        target_dataframe_name="X",
+        trans_primitives=["absolute"],
+        ignore_columns={"X": ["target"]},
+        seed_features=seed_features,
+    )
+    assert any(f.get_name() == "target" for f in features)
```

Comment:
probably not necessary but would we want to use `get_feature_names()` here as well for consistency?

Reply:
there's not really a need, imo, bc `target` is a base feature, so it's not possible for it to be a multi output feature (which are definitionally engineered features built from multi output primitives). It would just complicate the assertion here